src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -735,3 +735,21 @@ def _jdk_libs_paths_and_digest(self, hermetic_dist):
   @memoized_method
   def _jdk_libs_abs(self, nonhermetic_dist):
     return nonhermetic_dist.find_libs(self._JDK_LIB_NAMES)
+
+  def _double_check_cache_for_vts(self, vts, zinc_compile_context):
+    # Double check the cache before beginning compilation
+    if self.check_cache(vts):
+      self.context.log.debug(f'Snapshotting results for {vts.target.address.spec}')
+      classpath_entry = self._classpath_for_context(zinc_compile_context)
+      assert classpath_entry.directory_digest is None, f'Classpath entry {classpath_entry} from a double-check is expected to *not* have a digest set!'
```

Comment:
This seems like a strange thing to assert; would it make more sense to have this be an `if classpath_entry.directory_digest is None:` rather that an assertion that it is?

Reply:
I've changed the name of the function that's called to set the digest to `hydrate_missing_directory_digest()` to make it more clear why there's any checking being performed here. Essentially, I'm trying to assert that we are only snapshotting classpath entries that haven't previously been snapshotted, which is what we _expect_ from doing this as part of the "double check" process. I was trying to make sure we weren't getting any spurious invalidations from the v1 cache for classpath entries we'd already digested.

If using `Digest.load()` means we can avoid this sort of checking, we can probably remove it. But it's more just because this check is only necessary because certain classpath entries _aren't_ hydrated, so I wanted to make more sure of my existing hypothesis that the only unhydrated classpath entries are the ones that come in from the cache double check.